tests/test_controller/test_action_normalization.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (c) 2025 Robotics and AI Institute LLC. All rights reserved.
+
+from contextlib import nullcontext as does_not_raise
+
+import numpy as np
+
+from judo.controller import Controller, ControllerConfig
+from judo.optimizers import CrossEntropyMethod, CrossEntropyMethodConfig
+from judo.tasks import CylinderPush, CylinderPushConfig
+from judo.utils.normalization import IdentityNormalizer, MinMaxNormalizer, RunningMeanStdNormalizer
+
+# ##### #
+# TESTS #
+# ##### #
+
+
+def test_identity_normalizer() -> None:
+    """Test that IdentityNormalizer correctly passes through values."""
+    dim = 3
+    normalizer = IdentityNormalizer(dim)
+
+    # Test with random data
+    data = np.random.randn(10, dim)
+    normalized = normalizer.normalize(data)
+    denormalized = normalizer.denormalize(normalized)
+
+    # Check that the data is unchanged
+    np.testing.assert_array_almost_equal(data, normalized)
+    np.testing.assert_array_almost_equal(data, denormalized)
+
+
+def test_min_max_normalizer() -> None:
+    """Test that MinMaxNormalizer correctly scales values to [-1, 1] range."""
+    dim = 3
+    rand_vals = np.random.randn(dim, 2)
+    min_vals = np.min(rand_vals, axis=1)
+    max_vals = np.max(rand_vals, axis=1)
+
+    normalizer = MinMaxNormalizer(dim, min_vals, max_vals)
+
+    # Normalize min and max values
+    min_normalized = normalizer.normalize(min_vals)
+    max_normalized = normalizer.normalize(max_vals)
+
+    # Should be close to -1 and 1 respectively
+    np.testing.assert_array_almost_equal(min_normalized, -np.ones_like(min_normalized))
+    np.testing.assert_array_almost_equal(max_normalized, np.ones_like(max_normalized))
+
+    # Test with random data
+    data = np.random.uniform(min_vals, max_vals, (10, dim))
+    normalized = normalizer.normalize(data)
+    denormalized = normalizer.denormalize(normalized)
+
+    # Check that denormaling the normalized data returns the original data
+    np.testing.assert_array_almost_equal(data, denormalized)
+
+    # Check that normalized values are in [-1, 1] range
+    assert np.all(normalized >= -1.0 - 1e-6)
+    assert np.all(normalized <= 1.0 + 1e-6)
+
+
+def test_running_mean_std_normalizer() -> None:
+    """Test that RunningMeanStdNormalizer correctly updates and normalizes."""
+    dim = 3
+    normalizer = RunningMeanStdNormalizer(dim)
+
+    # Test with random data
+    data = np.random.randn(10, dim)
+    normalizer.update(data)
+
+    # Check that statistics were updated correctly
+    assert normalizer.count == 10
+    np.testing.assert_array_almost_equal(normalizer.mean, np.mean(data, axis=0))
+    np.testing.assert_array_almost_equal(normalizer.std, np.std(data, axis=0))
```

Comment:
can you add also a for loop here that makes sure iterative calls correctly update the running statistics? that's the main point of welford's algorithm, so this test doesn't really stress test that part